src/components/AdminRegisterPage/index.jsx

```diff
@@ -19,18 +23,33 @@ const AdminRegisterPage = ({ match }) => {
     );
   }
 
-  const { enterpriseSlug } = match.params;
-  const isEnterpriseAdmin = isEnterpriseUser(user, ENTERPRISE_ADMIN);
-  if (isEnterpriseAdmin) {
-    // user is authenticated and has the ``enterprise_admin`` JWT role, so redirect user to
-    // account activation page to ensure they verify their email address.
-    return <Redirect to={`/${enterpriseSlug}/admin/register/activate`} />;
-  }
+  useEffect(() => {
+    const processEnterpriseAdmin = (enterpriseUUID) => {
+      const isEnterpriseAdmin = isEnterpriseUser(user, ENTERPRISE_ADMIN, enterpriseUUID);
+      if (isEnterpriseAdmin) {
+        // user is authenticated and has the ``enterprise_admin`` JWT role, so redirect user to
+        // account activation page to ensure they verify their email address.
+        history.push(`/${enterpriseSlug}/admin/register/activate`);
+      } else {
+        // user is authenticated but doesn't have the `enterprise_admin` JWT role; force a log out so their
+        // JWT roles gets refreshed. on their next login, the JWT roles will be updated.
+        const logoutRedirectUrl = getLogoutRedirectUrl(getProxyLoginUrl(enterpriseSlug));
+        global.location.href = logoutRedirectUrl;
+      }
+      return <EnterpriseAppSkeleton />;
+    };
+
+    LmsApiService.fetchEnterpriseBySlug(enterpriseSlug)
```

Comment:
We didn't find any way to get the enterpriseUUID from context or any other place?

Reply:
No